flake.nix

```diff
@@ -385,27 +385,24 @@
 
         # Build only deps, but with llvm-cov so `workspaceCov` can reuse them cached
         workspaceDepsCov = craneLib.buildDepsOnly (commonArgs // {
-          pname = commonArgs.pname + "-lcov";
+          pnameSuffix = "-lcov-deps";
           src = filterWorkspaceDepsBuildFiles ./.;
-          cargoBuildCommand = "cargo llvm-cov --workspace --profile $CARGO_PROFILE";
+          buildPhaseCargoCommand = "cargo llvm-cov --workspace --profile $CARGO_PROFILE --no-report";
+          cargoBuildCommand = "dontuse";
+          cargoCheckCommand = "dontuse";
           nativeBuildInputs = commonArgs.nativeBuildInputs ++ [ cargo-llvm-cov ];
           doCheck = false;
         });
 
-        workspaceCov = craneLib.cargoBuild (commonArgs // {
-          pname = commonArgs.pname + "-lcov";
+        workspaceCov = craneLib.buildPackage (commonArgs // {
```

Comment:
I got the "use `workspaceCov` instead of `workspaceTestCov`" part when working on it myself, but didn't know the difference between `cargoBuild` and `buildPackage` (or even considered it).